### ISSUE TITLE
Rule proposal: declaration-block-semicolon-newline-after

### ISSUE THREAD

**Issue by davidtheclark** [stylelint/stylelint#65]
The `-after` suffix is because I don't think a user would likely require a newline *before*.

The question of whether this kind of suffix is a suitable convention I also brought up here: https://github.com/stylelint/stylelint/issues/21#issuecomment-102209896

**Comment by MoOx** [comment]
I already saw people writing this

```css
body {
  ;background: red
  ;color: white
}
```

This is weird but it possible :)

**Comment by davidtheclark** [comment]
Ok ... but in that case they aren't using a declaration block semicolon right? What `before` would mean in my mind is something like this:

```css
body {
  background: red;
  color: white
;}
```

I mean, we *could* allow for it ...

**Comment by MoOx** [comment]
it's not just about the last semicolon right ? Not sure why in your example you only have the last semicolon in a weird place.

**Comment by davidtheclark** [comment]
Oh I thought it was ONLY about the last semicolon ... I think I misunderstood. Now I get it :)

**Comment by MoOx** [comment]
Oh no. It's me that didn't read correctly...
But maybe we can just get this rule `declaration-block-semicolon` with after & before ? (before might be space or newline, same for after)

**Comment by davidtheclark** [comment]
Yeah, that's what I was thinking. And if we have that rule AND we have `block-closing-brace-space` and `-newline` ---- do we even need the `declaration-block-semicolon` rules at all?

**Comment by MoOx** [comment]
Good point.

**Comment by davidtheclark** [comment]
Actually I don't see `declaraton-semicolon-space` or `declaration-semicolon-newline` on the list in #1 (?). I guess I propose that we add those and then delete the `declaration-block-semicolon` spacing rules.

**Comment by MoOx** [comment]
Make sense.

**Comment by jeddy3** [comment]
Agreed. You also have access to edit the list #1, so feel free to go ahead and make that change.

**Comment by davidtheclark** [comment]
Cool, will do. I've been tweaking the list already a little.
